frameworklauncher/src/main/java/com/microsoft/frameworklauncher/webserver/LauncherModule.java

```diff
@@ -65,15 +70,112 @@ private static LaunchClientType getLaunchClientType(
+    checkWritableAccess(hsr, null, null);
+  }
+
+  private void checkWritableAccess(
+      HttpServletRequest hsr, String frameworkName) throws Exception {
+    checkWritableAccess(hsr, frameworkName, null);
+  }
+
+  private void checkWritableAccess(
+      HttpServletRequest hsr, String frameworkName, UserDescriptor frameworkUser) throws Exception {
+    if (!conf.getWebServerAclEnable()) {
+      return;
+    }
+
+    AclConfiguration aclConf = requestManager.getLauncherRequest().getAclConfiguration();
+    UserDescriptor user = UserDescriptor.newInstance(
+        getName(() -> hsr.getHeader(WebCommon.REQUEST_HEADER_USER_NAME)));
+
+    if (frameworkName == null) {
+      checkNonFrameworkWritableAccess(user, aclConf);
+    } else {
+      validateFrameworkUserConsistency(user, frameworkUser);
+      String namespace = CommonValidation.validateAndGetNamespace(frameworkName);
+      checkFrameworkWritableAccess(user, namespace, aclConf);
+    }
+  }
+
+  private void checkNonFrameworkWritableAccess(
+      UserDescriptor user, AclConfiguration aclConf) throws Exception {
+    if (!getAdminUsers(aclConf).contains(user)) {
+      throw new AuthorizationException(
+          "This operation needs administrator privilege.");
+    }
+  }
+
+  private void validateFrameworkUserConsistency(
+      UserDescriptor user, UserDescriptor frameworkUser) throws Exception {
+    if (frameworkUser != null && !frameworkUser.equals(user)) {
+      throw new BadRequestException(String.format(
+          "The UserName specified in the FrameworkDescriptor [%s] is " +
+              "not the same as the one specified in the HttpRequestHeader [%s].",
+          frameworkUser, user));
+    }
+  }
+
+  private void checkFrameworkWritableAccess(
+      UserDescriptor user, String namespace, AclConfiguration aclConf) throws Exception {
+    if (!getEffectiveNamespaceAcl(namespace, aclConf).containsUser(user)) {
+      throw new AuthorizationException(String.format(
+          "User [%s] does not have the Writable Access to the Namespace [%s].",
+          user, namespace));
+    }
+  }
+
+  private AccessControlList getEffectiveNamespaceAcl(
+      String namespace, AclConfiguration aclConf) throws Exception {
+    AccessControlList namespaceAcl = new AccessControlList();
+
+    // 1. Add Predefined Acl
+    // 1.1 Admin Users can always write the Namespace
+    namespaceAcl.addUsers(getAdminUsers(aclConf));
+    // 1.2 User can always write the Namespace whose name is the same as User
+    namespaceAcl.addUser(UserDescriptor.newInstance(namespace));
+
+    // 2. Add Configured Acl
+    namespaceAcl.addUsers(aclConf.getNamespaceAcls().
+        getOrDefault(namespace, new AccessControlList()).getUsers());
+
+    return namespaceAcl;
+  }
+
+  private Set<UserDescriptor> getAdminUsers(
+      AclConfiguration aclConf) throws Exception {
+    Set<UserDescriptor> adminUsers = new HashSet<>();
+    adminUsers.add(statusManager.getLauncherStatus().getLoggedInUser());
```

Comment:
>getLoggedInUser [](start = 53, length = 15)

The Launcher user is always admin, same principle as YARN:

  private AccessControlList getAdminAclList(Configuration conf) {
    AccessControlList aclList = new AccessControlList(conf.get(
        YarnConfiguration.YARN_ADMIN_ACL,
        YarnConfiguration.DEFAULT_YARN_ADMIN_ACL));
    aclList.addUser(daemonUser.getShortUserName());
    return aclList;
  }